tools/build_file_generator/JavaClassAnalyzer.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2018 The StartupOS Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.startupos.tools.buildfilegenerator;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.google.startupos.common.FileUtils;
+import com.google.startupos.tools.buildfilegenerator.Protos.Import;
+import com.google.startupos.tools.buildfilegenerator.Protos.JavaClass;
+
+import javax.inject.Inject;
+
+public class JavaClassAnalyzer {
+  private FileUtils fileUtils;
+
+  @Inject
+  public JavaClassAnalyzer(FileUtils fileUtils) {
+    this.fileUtils = fileUtils;
+  }
+
+  public JavaClass getJavaClass(String filePath) throws IOException {
+    JavaClass.Builder result = JavaClass.newBuilder();
+
+    String fileContent = fileUtils.readFile(filePath);
+    String classname = getClassname(filePath);
+    result.setClassname(classname).setPackage(getPackage(fileContent, classname));
+    getImportLines(fileContent).forEach(line -> result.addImport(getImport(line)));
+
+    result.setIsTestClass(isTestClass(fileContent));
+    result.setHasMainMethod(hasMainMethod(fileContent));
+
+    return result.build();
+  }
+
+  private static String getClassname(String filePath) {
+    String[] parts = filePath.split("/");
+    return parts[parts.length - 1].replace(".java", "");
+  }
+
+  private static String getPackage(String fileContent, String classname) {
+    List<String> packageLines = getLinesStartWithKeyword(fileContent, "package", "");
+    if (packageLines.isEmpty()) {
+      throw new IllegalArgumentException(
+          String.format("Can't find package for the file: %s", classname));
+    }
+    if (packageLines.size() > 1) {
+      throw new IllegalArgumentException(
+          String.format("Found %d packages for the file: %s", packageLines.size(), classname));
+    }
+    return packageLines.get(0).split(" ")[1].replace(";", "");
+  }
+
+  private static List<String> getImportLines(String fileContent) {
+    return getLinesStartWithKeyword(fileContent, "import", ".");
+  }
+
+  private static List<String> getLinesStartWithKeyword(
+      String fileContent, String keyword, String lineShouldContain) {
+    return Arrays.stream(fileContent.split(System.lineSeparator()))
+        .map(String::trim)
+        .collect(Collectors.toList())
```

Comment:
Do we need to collect and then stream again? What happens if we don't?

Reply:
Yes, you are right. It isn't necessarily. 
The redundant call deleted.